src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -37,13 +37,10 @@ public static partial class JsonSerializer
 
                 state.Current.KeyName = reader.GetString();
             }
-            else if (state.Current.JsonClassInfo.ClassType == ClassType.Value)
-            {
-                // We should be in a converter, thus we must have bad JSON.
```

Comment:
With the proper checks implemented in `HandleStartObject` and `HandleStartDictionary` (which immediately precede the invocation of this method), we should no longer end up in this state.

One way we could have gotten here prior to this change is deserialization cases like this:

```c#
var json = JsonSerializer.Deserialize<int[]>("{ \"a\": \"b\" }")
```

Without the following guard, we would push the `ClassType.Value` JsonClassInfo (representing the `int` element type) to the read stack, which would allow the removed condition to be true.